### PR TITLE
feat(servergroups): Add endpoint to retrieve server groups by applications

### PR DESCRIPTION
**Pull request by jeyrschabu**
- Takes a list of applications and return an aggregated summary

### CODE REVIEW

**Comment by jeyrschabu** [comment]
Convenient way to get a list of server groups by applications. depends on https://github.com/spinnaker/gate/pull/426

@spinnaker/netflix-reviewers PTAL

**Comment by anotherchrisberry** [comment]
A couple of things:
1. it might be nice to group the response by application, although I'm not sure of the particular use case this endpoint has in mind, so maybe leaving them ungrouped is totally fine.
2. probably worth figuring out the `@PreAuthorize` story or adding a `@PostAuthorize` mechanism to filter out applications the user should not be able to access

**Comment by robzienert** [comment]
RE @anotherchrisberry item 1: This is severe pedantry, but... I'd tend to disagree with grouping by application. The response is supposed to be a list of server groups. I'd be surprised to get a pre-grouped response back. Maybe worth a query param to alter the response structure?

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy [hunk 1]
```diff
@@ -140,6 +138,15 @@ class ServerGroupController {
     return summaryList(application, cloudProvider)
   }
 
+  @PostAuthorize("@authorizationSupport.filterForAccounts(returnObject)")
+  @RequestMapping(value= "/serverGroups", method = RequestMethod.GET)
+  List getServerGroupsByApplications(@RequestParam(value = 'applications') List<String> applications,
+                                     @RequestParam(required = false, value = 'cloudProvider') String cloudProvider) {
+    def result = []
+    applications.forEach { application -> result +=summaryList(application, cloudProvider)}
```
Comment (robzienert) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy]: Could make the entire method body `applications.collectMany { app -> summaryList(application, cloudProvider) }`

**Comment by ajordens** [comment]
I'd also argue against a query parameter that alters the response structure ... any sort of grouping can easily be done client-side.

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy (hunk 1 above)
Comment (jeyrschabu) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy]: fancy

**Comment by robzienert** [comment]
Suppose response altering would make gRPC even harder. +1

**Comment by jeyrschabu** [comment]
sounds like a strong consensus on not grouping the response. I will address item #2

**Pull request merged**